packages/compile-solidity/src/normalizeOptions.ts

```diff
@@ -10,10 +10,17 @@ export const normalizeOptions = (options: Config) => {
   options.compilers.solc.settings.evmVersion =
     options.compilers.solc.settings.evmVersion ||
     options.compilers.solc.evmVersion;
-  options.compilers.solc.settings.optimizer =
-    options.compilers.solc.settings.optimizer ||
-    options.compilers.solc.optimizer ||
-    {};
+
+  if (options.compilers.solc.disableDefaults) {
+    // if the user has disabled defaults, we don't want to set optimizer defaults; Solidity versions prior to 0.8.6 will produce different bytecodes if optimizer settings are missing vs. set to false; this matters for how Blockscout does verification
+    options.compilers.solc.settings.optimizer =
+      options.compilers.solc.settings.optimizer || {};
+  } else {
+    options.compilers.solc.settings.optimizer =
+      options.compilers.solc.settings.optimizer ||
+      options.compilers.solc.optimizer ||
+      {};
+  }
```

Comment:
The grandfathered check below also sets the optimizer without the disableDefaults guard. Is this the desired behavior?

Reply:
I considered this and decided that this is a very specific edge case, using a previous way we configured compilers and shouldn't really be relevant here. If you or @gnidan think otherwise, let me know how you'd like it handled.

Reply:
Seems worth having the old-style config respect the new `disableDefaults` flag, since we don't know how prevalent this old-style is, and since it's not really that much of a lift to make it respect this new flag.

Reply:
Well, looking at it -- the optimizer setting _already_ doesn't get set to a default, right? It will always be `options.solc.optimizer` for the grandfathered case. Does that require a change?

Reply:
Hm you might be right! Guess the workaround this whole time was to use the old-style solc config!

Reply:
Are old-style solc config options a viable work-around for blockscout?

Reply:
Could be wrong but it does seem that way. Not that I think we should lean on that since this is already so far along

Reply:
One _could_ probably use this as a workaround. Just set `options.solc.optimizer` to `options.solc.settings.optimizer` when fetching from Blockscout? Wouldn't require this PR at all in that case, right. But I think that is a less straightforward hack than the one proposed here. Not to mention that adding default optimizer settings is itself incorrect, so this PR would allow us to rectify it for Blockscout contracts without breaking anything.

Reply:
is there a way to have a `normalizeForBlockscout` function that is only invoked for blockscout needed resolution?

Reply:
I'd be fine with that! When I asked @gnidan he wanted this to be a user-facing setting, though I can see now that may have been due to our misunderstanding around how broad/narrow it would be. If he's ok with it, I'd be happy to refactor in this way.

Reply:
> is there a way to have a `normalizeForBlockscout` function that is only invoked for blockscout needed resolution?

Ugh please no. Everyone's always talking about the amount of tech. debt we have these days. We must be mindful of these sorts of hundred-line PRs... they should not introduce new debt!